src/libcore/yk_swt_impl.c

```diff
@@ -87,17 +87,13 @@ yk_swt_rec_loc_impl(uint64_t crate_hash, uint32_t def_idx, uint32_t bb_idx)
 
 
 // Stop tracing on the current thread.
-// The current thread must already be tracing. The trace buffer is returned
-// and the number of locations it holds is written to `*ret_trace_len`. It is the
-// responsibility of the caller to free the trace buffer. A NULL pointer is
-// returned if the trace was invalidated.
+// On success the trace buffer is returned and the number of locations it
+// holds is written to `*ret_trace_len`. It is the responsibility of the caller
+// to free the trace buffer. A NULL pointer is returned on error (if tracing
+// wasn't started, or the trace was invalidated).
```

Comment:
Maybe add "Calling this function without having previously called yk_swt_start_tracing_impl will lead to undefined behaviour."?

Reply:
Not true. I've designed it in such a way that it would return NULL reliably.

Reply:
> you actually have to do this

Well, you don't *have* to, but why you wouldn't I don't know.

> plus it gives us a get-out-of-jail-free card in the future if we need to change things

See you should have started with this argument :)

Reply:
Ha!